### ISSUE TITLE
Kane's exclusions to discounts based on promotions

### ISSUE THREAD

**Issue by kbower** [kbower/tickettest1#1736]
_Imported from trac ticket 1736.
Created by **wayne**
Opened in trac: Jun 28, 2012, 11:18:07 AM
Last modified in trac: Jul 2, 2012, 1:44:13 PM_
***
_by wayne:_
Kent,
If any item has a promotional price from promotion code AVP, AVY or AVZ then discount codes 10S, 4TH , IND and I40 cannot be applied to those items. In addition, Items on promotion code T can have discount codes 10S, 4TH , IND and I40 applied to their promotional price.
Thanks,
*Wayne Liburd*

**Comment by kbower** [comment]
_**kbower** trac comment on Jun 28, 2012, 11:23:12 AM:_

Wayne, you said 3 promotion codes cannot be applied and 1 can be.  What about the remaining 514?

**Comment by kbower** [comment]
_**kbower** trac comment on Jun 28, 2012, 12:14:01 PM:_

Kent,

These are the only promotions that will be active so you can exclude all promotions except 'T'.

Thanks,

*Wayne Liburd*

**Comment by kbower** [comment]
_**kbower** trac comment on Jun 28, 2012, 12:18:06 PM:_

Ok, but the root of the problem is that there is no effective date range in ITM$PROM_PRC_TP.

If you use ITM$RET_CLNDR instead then we could work this out.  Or if you put records in ITM$PROM_PRC_TP /and never delete them./

Can either of these be worked out?

**Comment by kbower** [comment]
_**kbower** trac comment on Jul 2, 2012, 11:42:30 AM:_

Is this something you want as a *standard exclusion*, or just those 4 discounts: 10S, 4TH, IND, I40?

Also, be aware of the same trap we just experienced if the *package* is on promotion but the *item* is not.

Kent

**Comment by kbower** [comment]
_**kbower** trac comment on Jul 2, 2012, 11:56:34 AM:_

Kent,

The standard would exclude all promotions but in this case we are allowing promotion code T and YA.


Thanks,
Wayne Liburd
----
You never mentioned 'YA'.

Also, there is no such standard exclusion that excludes promotions except for RT.

**Comment by kbower** [comment]
_**kbower** trac comment on Jul 2, 2012, 1:01:50 PM:_

(Changeset [[2808]](../commit/6cd33554a99c37c33328a2a8bcbed6ee4acbc832))
 * Kane's exclusions to discounts based on promotions (closes #1736)

**Comment by kbower** [comment]
_**kbower** trac status change on Jul 2, 2012, 1:01:50 PM:_ **closed**

**Comment by kbower** [comment]
_**kbower** trac comment on Jul 2, 2012, 1:44:13 PM:_

Kent,

You are right I did not mention YA before but I would want it allowed for those four codes if we can do it.

I am aware there is no such standard exclusion because we did not have that option before. However, if we have special discounts in an insert we generally do not give additional discounts and is now manually accommodated by the salespeople.

Thanks,

*Wayne Liburd*
